internal/server/singleprocess/service_project.go

```diff
@@ -94,3 +103,38 @@ func (s *service) UpsertApplication(
 
 	return &pb.UpsertApplicationResponse{Application: app}, nil
 }
+
+func queueInitOps(s *service, ctx context.Context, project *pb.Project) error {
+	workspaces, err := s.state.WorkspaceList()
+
+	if err != nil {
+		return err
+	}
+
+	if len(workspaces) == 0 {
+		workspaces = append(workspaces, &pb.Workspace{Name: "default"})
+	}
+
+	for _, workspace := range workspaces {
```

Comment:
Ah yes, workspace-scoped operations are also a thing that is going to be an issue quickly. 😄

Reply:
Indeed